### PR TITLE
[MRG] Relative entropy policy search

### PR DESCRIPTION
**Pull request by AlexanderFabisch**

### CODE REVIEW

**Comment by AlexanderFabisch** [comment]
@HWiese1980 this is one dependency that we have, needs review

File: bolero/optimizer/reps.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+        self.history_theta = deque(maxlen=self.n_samples_per_update)
+        self.history_R = deque(maxlen=self.n_samples_per_update)
+
+    def get_next_parameters(self, params, explore=True):
+        """Return parameter vector that shall be evaluated next.
+
+        Parameters
+        ----------
+        params : array-like
```
Comment (maotto) [inline bolero/optimizer/reps.py]: the array probably needs to have a specific shape; I suggest adding ", shape = (num_params,)"

File: bolero/representation/test/test_ul_policies.py [hunk 2]
```diff
@@ -30,6 +31,44 @@ def test_linear_gaussian():
     assert_almost_equal(np.std(Y_sampled), 1.0, places=1)
 
 
+def test_constant_gaussian_full_covariance():
+    random_state = np.random.RandomState(0)
+
+    n_samples = 10000
+    n_weights = 5
+    mean = np.ones(n_weights)
+    ulp = ConstantGaussianPolicy(
```
Comment (maotto) [inline bolero/representation/test/test_ul_policies.py]: ulp doesn't tell much, I would prefer a longer name

File: bolero/representation/test/test_ul_policies.py [hunk 3]
```diff
@@ -30,6 +31,44 @@ def test_linear_gaussian():
...
+        n_weights, covariance="full", mean=mean,
+        covariance_scale=1.0, random_state=random_state)
+    Y = mean + random_state.randn(n_samples, n_weights)
+    ulp.fit(None, Y, np.ones(n_samples))
+    estimated_mean = ulp(explore=False)
+    assert_array_almost_equal(mean, estimated_mean, decimal=2)
+
+    p = ulp.probabilities([mean])
+    p2 = ulp.probabilities([mean + 1.0])
```
Comment (maotto) [inline bolero/representation/test/test_ul_policies.py]: same here, variables p and p2 should have names indicating what probabilities they store

File: bolero/optimizer/reps.py [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+        self.log_to_stdout = log_to_stdout
+        self.random_state = random_state
+
+    def init(self, n_params):
+        """Initialize optimizer.
+
+        Parameters
+        ----------
+        n_params : int
+            dimension of the parameter vector
```
Comment (maotto) [inline bolero/optimizer/reps.py]: dimensionality

File: bolero/representation/ul_policies.py [hunk 5]
```diff
@@ -156,6 +156,132 @@ def fit(self, X, Y, weights=None, context_transform=True):
...
+        Y : array-like, shape (num_samples, n_weights)
+            2d array of parameter vectors
+
+        Returns
+        ----------
+        resp : array-like, shape (num_samples)
+            the probabilities of the samples under this policy
+
+        """
+        # TODO: Check SciPy version (>= 0.14 for multivariate_normal)
```
Comment (maotto) [inline bolero/representation/ul_policies.py]: doable with little effort: https://stackoverflow.com/a/21385217

**Pull request merged**